modules/terrain/src/lib/parse-terrain.js

```diff
@@ -89,15 +109,49 @@ function getMartiniTileMesh(terrainImage, terrainOptions) {
       boundingBox: getMeshBoundingBox(attributes)
     },
     mode: 4, // TRIANGLES
-    indices: {value: triangles, size: 1},
+    indices: {value: Uint32Array.from(triangles), size: 1},
     attributes
   };
 }
 
+/**
+ * Get Martini generated vertices and triangles
+ *
+ * @param {number} meshMaxError threshold for simplifying mesh
+ * @param {number} width width of the input data
+ * @param {*} terrain elevation data
+ * @returns generated vertices and triangles
+ */
+function getMartiniTileMesh(meshMaxError, width, terrain) {
+  const gridSize = width + 1;
+  const martini = new Martini(gridSize);
+  const tile = martini.createTile(terrain);
+  const {vertices, triangles} = tile.getMesh(meshMaxError);
+
+  return {vertices, triangles};
+}
+
+/**
+ * Get Delatin generated vertices and triangles
+ *
+ * @param {number} meshMaxError threshold for simplifying mesh
+ * @param {number} width width of the input data array
+ * @param {number} height height of the input data array
+ * @param {*} terrain elevation data
+ * @returns generated vertices and triangles
+ */
+function getDelatinTileMesh(meshMaxError, width, height, terrain) {
+  const tin = new Delatin(terrain, width + 1, height + 1);
```

Comment:
- Does Delating throw a good user facing exception if the dimensions are wrong, or should we check for this before we call martini?

Reply:
Delatin doesn't make any dimension restriction. Only Martini throws if dimensions aren't a square power of 2.